x-pack/plugin/esql/qa/testFixtures/src/main/resources/date.csv-spec

```diff
@@ -614,3 +614,60 @@ birth_date:datetime
 1952-02-27T00:00:00.000Z
 1953-04-21T00:00:00.000Z
 ;
+
+dateMathAddDatePeriodsFromLeft
+row n = to_dt("1953-04-04T00:00:00.000Z") | eval then = 4 years + 3 months + 2 weeks + 1 day + n | keep then;
+
+then:datetime
+1957-07-19T00:00:00.000Z
+;
+
+dateMathAddDatePeriodsThenAddToDatetime
+row n = to_dt("1953-04-04T00:00:00.000Z") | eval then = (4 years + 3 months + 2 weeks + 1 day) + n | keep then;
+
+then:datetime
+1957-07-19T00:00:00.000Z
+;
+
+dateMathSubtractDatePeriodsFromLeft
+row n = to_dt("1957-07-19T00:00:00.000Z") | eval then = -4 years - 3 months - 2 weeks - 1 day + n | keep then;
```

Comment:
The second line will not work exactly like this because `z = -2 weeks - 1 day` tries to assign an unrepresentable value; I did `then = y + (-2 weeks -1 day)` instead.

Reply:
> The second line will not work exactly like this because z = -2 weeks - 1 day
That calls for another test that handles negation of date intervals.

Reply:
That is a very good remark. I thought negations were already handled and tested fine (there's multiple tests which contain expressions like `-1hour` etc.), but it turns out that negation fails when used over a sum like `-(1hour + 2hour)`.

I'll take care of that.

Reply:
So, handling expressions like `now() + (-(1hour + 2hour))` will require increasing the scope of this endeavor to generalize folding of `Neg` as well, together with associated unit and csv tests. I'd like to do this in a follow-up PR because that for sure warrants a separate review.